src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -135,15 +135,23 @@ const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
 
   // Handling possibly incorrectly (by MP) possible null values
   const suggestedAddresses =
-    verifiedAddressResult.suggestedAddresses ||
+    verifiedAddressResult.verifyAddressOrError?.suggestedAddresses ||
     ([] as NonNullable<
-      NonNullable<AddressVerificationFlow_verifiedAddressResult$data>
+      NonNullable<
+        NonNullable<AddressVerificationFlow_verifiedAddressResult$data>
+      >["verifyAddressOrError"]
     >["suggestedAddresses"])
-  const inputAddress = verifiedAddressResult.inputAddress as NonNullable<
-    NonNullable<AddressVerificationFlow_verifiedAddressResult$data>
+  const inputAddress = verifiedAddressResult.verifyAddressOrError
+    ?.inputAddress as NonNullable<
+    NonNullable<
+      NonNullable<AddressVerificationFlow_verifiedAddressResult$data>
+    >["verifyAddressOrError"]
   >["inputAddress"]
-  const verificationStatus = verifiedAddressResult.verificationStatus as NonNullable<
-    AddressVerificationFlow_verifiedAddressResult$data
+  const verificationStatus = verifiedAddressResult.verifyAddressOrError
+    ?.verificationStatus as NonNullable<
+    NonNullable<
+      AddressVerificationFlow_verifiedAddressResult$data
+    >["verifyAddressOrError"]
```

Comment:
I know this code was mostly all here before, but do all of these verbose types need to be repeated? Also, `suggestedAddresses` is different than `inputAddress` and `verificationStatus`, so casting in this way (seems) to be incorrect. 

As `verifiedAddressResult.verifyAddressOrError` is a union underneath, you can use the `Extract` typescript util helper to pull out the desired "slice" of the union:

```tsx
type SuccessType = Extract<SomeType, { __typename: 'SomeTypeSuccess' }>
type ErrorType = Extract<SomeType, { __typename: 'SomeTypeError' }>

const foo: SuccessType = ...
```

This might help here, but not too sure!

Reply:
Thanks for the review.
Those are being handled at https://github.com/artsy/force/pull/12689/files, so I will wait until that's merged to do a rebase here.